tests/fields/test_BaseField.py

```diff
@@ -61,3 +64,40 @@ def test_render_metadata():
     expected = [("encoding", "some_exotic_encoding"), ("doc", "Official Breed Name")]
 
     assert expected == field.get_metadata()
+
+
+def test_render_complex_types():
+    @dataclasses.dataclass
+    class Metadata(AvroModel):
+        timestamp: datetime = dataclasses.field(
+            default_factory=lambda: datetime(2023, 10, 21, 11, 11),
+        )
+
+    parent = AvroModel()
+    parent.metadata = utils.SchemaMetadata.create(type)
+    field = AvroField(
+        "metadata",
+        Metadata,
+        metadata={"desc": "Some metadata"},
+        default_factory=Metadata,
+        parent=parent,
+    )
+
+    expected = {
+        "desc": "Some metadata",
+        "name": "metadata",
+        "type": {
+            "type": "record",
+            "name": "Metadata",
+            "fields": [
+                {
+                    "name": "timestamp",
+                    "type": {"type": "long", "logicalType": "timestamp-millis"},
+                    "default": 1697886660000,
+                }
+            ],
+        },
+        "default": {"timestamp": "2023-10-21T11:11:00"},
```

Comment:
Should this also be an epoch time? 🤔

Reply:
I did change the behaviour to call default_to_avro recursively on fields of a record. that way we render flat fields eventually.